src/index.js

```diff
@@ -92,49 +92,52 @@ IdTokenVerifier.prototype.verify = function(token, nonce, cb) {
   var nbf = jwt.payload.nbf;
   var tnonce = jwt.payload.nonce || null;
   /* eslint-enable vars-on-top */
+  var _this = this;
 
-  if (this.issuer !== iss) {
-    return cb(
-      new error.TokenValidationError('Issuer ' + iss + ' is not valid.'),
-      false
-    );
-  }
-
-  if (this.audience !== aud) {
-    return cb(
-      new error.TokenValidationError('Audience ' + aud + ' is not valid.'),
-      false
-    );
-  }
+  this.getRsaVerifier(iss, kid, function(err, rsaVerifier) {
+    if (err) {
+      return cb(err);
+    }
+    if (rsaVerifier.verify(headAndPayload, signature)) {
+      if (_this.issuer !== iss) {
+        return cb(
+          new error.TokenValidationError('Issuer ' + iss + ' is not valid.'),
+          false
+        );
+      }
 
-  if (this.expectedAlg !== alg) {
-    return cb(
-      new error.TokenValidationError(
-        'Algorithm ' +
-          alg +
-          ' is not supported. (Expected algs: [' +
-          supportedAlgs.join(',') +
-          '])'
-      ),
-      false
-    );
-  }
+      if (_this.audience !== aud) {
+        return cb(
+          new error.TokenValidationError('Audience ' + aud + ' is not valid.'),
+          false
+        );
+      }
 
-  if (tnonce !== nonce) {
-    return cb(new error.TokenValidationError('Nonce does not match.'), false);
-  }
+      if (tnonce !== nonce) {
+        return cb(
+          new error.TokenValidationError('Nonce does not match.'),
+          false
+        );
+      }
 
-  var expirationError = this.verifyExpAndNbf(exp, nbf); // eslint-disable-line vars-on-top
+      if (_this.expectedAlg !== alg) {
```

Comment:
If this whole code is being executed as part of the result of an "RSAVerification", in what case would this if make sense, given that the verification is already done and either succeeded or failed? A bit late to say "algorithm not supported"